### PR TITLE
fix: non-deterministic evm execution result

### PR DESCRIPTION
**Pull request by yihuang**
## Description

When there's extra-eips, concurrent query could cause non-deterministic evm execution result. 
It could affect even if extra-eips only exists in historical states.

Ref: https://github.com/ethereum/go-ethereum/pull/26137

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -79,6 +79,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (state) [#1320](https://github.com/evmos/ethermint/pull/1320) Fix codehash check mismatch when the code has been deleted in the evm state.
 * (rpc) [#1392](https://github.com/evmos/ethermint/pull/1392) Allow fill the proposer address in json-rpc through tendermint api, and pass explicitly to grpc query handler.
 * (rpc) [#1431](https://github.com/evmos/ethermint/pull/1431) Align hex-strings proof fields in `eth_getProof` as Ethereum.
+* (evm) [#]() Fix non-deterministic evm execution when enable-eips is not empty.
```
Comment (yihuang) [inline CHANGELOG.md:82]:
```suggestion
* (evm) [#1436](https://github.com/evmos/ethermint/pull/1436) Fix non-deterministic evm execution when enable-eips is not empty.
```

Comment (yihuang) [inline CHANGELOG.md:82]:
```suggestion
* (evm) [#1436](https://github.com/evmos/ethermint/pull/1436) Fix non-deterministic evm execution when enable-eips is not empty.
```

**Review by adisaran64 — COMMENTED** [review]
@yihuang looks like integration tests are failing. could you take a look?

File: go.mod [hunk 2]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/cosmos/go-bip39 v1.0.0
 	github.com/cosmos/ibc-go/v5 v5.0.1
 	github.com/davecgh/go-spew v1.1.1
-	github.com/ethereum/go-ethereum v1.10.19
+	github.com/ethereum/go-ethereum v1.10.24-0.20221109065213-7dc5e785a8fd
```
Comment (fedekunze) [inline go.mod:14]: needs update to latest go-ethereum

Comment (yihuang) [reply go.mod:14]: it's in the main branch, do you mean update to current main, or wait for a point release.

Comment (yihuang) [reply go.mod:14]: Do you mean update to latest go-ethereum directly will break things?

Comment (fedekunze) [reply go.mod:14]: yes, it's breaking. See #1254

Comment (yihuang) [reply go.mod:14]: I see, maybe we close this PR and prioritize the #1254

**Pull request closed without merging**